cmd/kubeadm/app/discovery/token/token.go

```diff
@@ -205,8 +209,17 @@ func runForEndpointsAndReturnFirst(endpoints []string, fetchKubeConfigFunc func(
 			}, constants.DiscoveryRetryInterval, stopChan)
 		}(endpoint)
 	}
-	wg.Wait()
-	return resultingKubeConfig
+	select {
+	case <-time.After(constants.DiscoveryTimeout):
```

Comment:
Why do we hard const this vs. conf variable?

Reply:
Providing an extra flag?

Reply:
I think more of a config variable, rather than extra flag. If you are both OK with that I'll do it that way.

Reply:
A configurable variable seems okay to me.

Reply:
configuration variable makes more sense.